test/unit/stream_alert_alert_processor/test_output_base.py

```diff
@@ -0,0 +1,196 @@
+'''
+Copyright 2017-present, Airbnb Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+import json
+import os
+import urllib2
+
+import boto3
+
+from mock import patch, Mock
+
+from moto import mock_s3, mock_kms
+from nose.tools import assert_equal, assert_is_not_none
+
+from stream_alert.alert_processor.main import (
+    _load_output_config as load_config,
+    _sort_dict
+)
+
+from stream_alert.alert_processor.output_base import (
+    OutputProperty,
+    StreamOutputBase
+)
+
+from unit.stream_alert_alert_processor import (
+    REGION,
+    FUNCTION_NAME,
+    CONFIG
+)
+
+from unit.stream_alert_alert_processor.helpers import (
+    _encrypt_with_kms,
+    _put_mock_creds,
+    _put_s3_test_object
+)
+
+# Remove all abstractmethods from __abstractmethods__ so we can
+# instantiate StreamOutputBase for testing
+StreamOutputBase.__abstractmethods__ = frozenset()
+StreamOutputBase.__service__ = 'test_service'
+
+def test_output_property_default():
+    """OutputProperty defaults"""
+    prop = OutputProperty()
+
+    assert_equal(prop.description, '')
+    assert_equal(prop.value, '')
+    assert_equal(prop.input_restrictions, {' ', ':'})
+    assert_equal(prop.mask_input, False)
+    assert_equal(prop.cred_requirement, False)
+
+
+class TestSteamOutputBase(object):
+    """Test class for StreamOutputBase"""
```

Comment:
Pls make this more descriptive, these are tests common across all Outputs right?